libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1023,35 +1023,42 @@ std::string YulUtilFunctions::resizeArrayFunction(ArrayType const& _type)
 					sstore(array, newLen)
 				</isDynamic>
 
-				// Size was reduced, clear end of array
-				if lt(newLen, oldLen) {
-					let oldSlotCount := <convertToSize>(oldLen)
-					let newSlotCount := <convertToSize>(newLen)
-					let arrayDataStart := <dataPosition>(array)
-					let deleteStart := add(arrayDataStart, newSlotCount)
-					let deleteEnd := add(arrayDataStart, oldSlotCount)
-					<?packed>
-						// if we are dealing with packed array and offset is greater than zero
-						// we have  to partially clear last slot that is still used, so decreasing start by one
-						let offset := mul(mod(newLen, <itemsPerSlot>), <storageBytes>)
-						if gt(offset, 0) { <partialClearStorageSlot>(sub(deleteStart, 1), offset) }
-					</packed>
-					<clearStorageRange>(deleteStart, deleteEnd)
-				}
-			})")
-			("functionName", functionName)
-			("panic", panicFunction())
-			("fetchLength", arrayLengthFunction(_type))
-			("isDynamic", _type.isDynamicallySized())
-			("convertToSize", arrayConvertLengthToSize(_type))
-			("dataPosition", arrayDataAreaFunction(_type))
-			("clearStorageRange", clearStorageRangeFunction(*_type.baseType()))
-			("maxArrayLength", (u256(1) << 64).str())
-			("packed", _type.baseType()->storageBytes() <= 16)
-			("itemsPerSlot", to_string(32 / _type.baseType()->storageBytes()))
-			("storageBytes", to_string(_type.baseType()->storageBytes()))
-			("partialClearStorageSlot", partialClearStorageSlotFunction())
-			.render();
+				<?needsClearing>
+					// Size was reduced, clear end of array
+					if lt(newLen, oldLen) {
+						let oldSlotCount := <convertToSize>(oldLen)
+						let newSlotCount := <convertToSize>(newLen)
+						let arrayDataStart := <dataPosition>(array)
+						let deleteStart := add(arrayDataStart, newSlotCount)
+						let deleteEnd := add(arrayDataStart, oldSlotCount)
+						<?packed>
+							// if we are dealing with packed array and offset is greater than zero
+							// we have  to partially clear last slot that is still used, so decreasing start by one
+							let offset := mul(mod(newLen, <itemsPerSlot>), <storageBytes>)
+							if gt(offset, 0) { <partialClearStorageSlot>(sub(deleteStart, 1), offset) }
+						</packed>
+						<clearStorageRange>(deleteStart, deleteEnd)
+					}
+				</needsClearing>
+			})");
+			templ("functionName", functionName);
+			templ("maxArrayLength", (u256(1) << 64).str());
+			templ("panic", panicFunction());
+			templ("fetchLength", arrayLengthFunction(_type));
+			templ("isDynamic", _type.isDynamicallySized());
+			bool isMappingBase = _type.baseType()->category() == Type::Category::Mapping;
+			templ("needsClearing", !isMappingBase);
```

Comment:
Could just inline it here

Reply:
Ah no, also used below

Reply:
Weird, I thought this was somehow done before, as part of the `zeroValueFunction` or something like that

Reply:
Hm, there is an assertion in zeroValueFunction.
I wanted to do it this way as it was doing the same in old codegen. But, other option is to implement noop zero function for `MappingType` which would be maybe weird...